src/Common/src/NativeMethods.cs

```
@@ -4666,8 +4596,8 @@ public class EVENTMSG
         [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
         public interface IProvideClassInfo
         {
-            [return: MarshalAs(UnmanagedType.Interface)]
-            UnsafeNativeMethods.ITypeInfo GetClassInfo();
+            [PreserveSig]
+            Interop.HRESULT GetClassInfo(out UnsafeNativeMethods.ITypeInfo ppTI);
```

I presume we never called this method, if so this whole interface could go and we can use `IntPtr` or `void*` where ever this interface was used.

---

It is used in `COM2TypeInfoProcessor.cs`

```cs
public static UnsafeNativeMethods.ITypeInfo FindTypeInfo(object obj, bool wantCoClass)
{
    UnsafeNativeMethods.ITypeInfo pTypeInfo = null;

    // This is kind of odd.  What's going on here is that if we want the CoClass (e.g. for
    // the interface name), we need to look for IProvideClassInfo first, then look for the
    // typeinfo from the IDispatch. In the case of many OleAut32 operations, the CoClass
    // doesn't have the interface members on it, although in the shell it usually does, so
    // we need to re-order the lookup if we *actually* want the CoClass if it's available.

    for (int i = 0; pTypeInfo == null && i < 2; i++)
    {
        if (wantCoClass == (i == 0))
        {
            if (obj is NativeMethods.IProvideClassInfo pProvideClassInfo)
            {
                try
                {
                    pProvideClassInfo.GetClassInfo(out pTypeInfo);
                }
                catch
                {
                }
            }
        }
        else
        {
            if (obj is UnsafeNativeMethods.IDispatch iDispatch)
            {
                try
                {
                    iDispatch.GetTypeInfo(0,Kernel32.GetThreadLocale(), out pTypeInfo);
                }
                catch
                {
                }
            }
        }
    }

    return pTypeInfo;
}
```

---

My mistake, I didn't see that you changed it to `[PreserveSig]`. You should either change the signature back and remove the `[return]` or check the HRESULT and remove the `try..catch` blocks. Since we're ignoring everything I'd prefer to remove the catch all blocks- since generally they're evil.